lib/adbc_driver.ex

```diff
@@ -225,7 +247,7 @@ defmodule Adbc.Driver do
     {:ok, zip_files} = :zip.table(cache_path)
 
     for {:zip_file, filename, _, _, _, _} <- zip_files,
-        Path.extname(filename) == ".so" do
+        Path.extname(filename) in [".so", ".dylib"] do
       {:ok, {filename, file_data}} = :zip.zip_get(filename, zip_handle)
 
       filename =
```

Comment:
I think instead of calling String.replace_leading to build the filename, we should just call `adbc_driver_so`. Then we can keep the same filename for duckdb too, we just need to change the extension.